refactor(core): convert FlowConcurrency.vue to TypeScript with Composition API

mohammadshahidbeigh opened the pull request:
Closes https://github.com/kestra-io/kestra/issues/12080.

**_This PR converts FlowConcurrency.vue to use TypeScript and the Composition API_**. 


1. Converted FlowConcurrency.vue to Composition API with TypeScript using <script setup lang="ts">, preserved the template and styles, and set inheritAttrs: false via defineOptions.
2. Introduced typed refs and computeds (runningCount, totalCount, runningCountSet, progress) and a typed setRunningCount handler.
3. Replaced direct flowStore.flow access in the template with safe computeds: hasConcurrency, limit, behavior, namespace, and flowId. Added currentFlow with a local FlowLike type to satisfy TS.
4. Resolved all linter/type errors for this file.

mohammadshahidbeigh committed:
Convert FlowConcurrency.vue to TypeScript with Composition API

MilosPaunovic commented:
Just a small note for the future, if you could use **GitHub Keywords** when opening PRs to connect it to the proper issue (to automate the entire process a bit more), it would be super helpful to us. You can read a bit more about it [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue) (I've added closing keyword for this particular issue).

mohammadshahidbeigh commented:
Got it, thanks for the tip! I’ll make sure to use GitHub keywords in future PRs to properly link them to the related issues. Appreciate the guidance! 🙏

MilosPaunovic commented on ui/src/monaco-editor.d.ts:
Changes in this file need to be reverted, as they're not relevant to the issue at hand.

mohammadshahidbeigh replied:
Hi @MilosPaunovic,
Got it — I’ll revert the changes in the ui/src/monaco-editor.d.ts file and push the update shortly. Thanks for pointing that out! 🙏

mohammadshahidbeigh replied:
Hi @MilosPaunovic,
I’ve reverted the changes in the ui/src/monaco-editor.d.ts file as requested. Thanks for the guidance! 🙏

MilosPaunovic replied:
If you did handle what was requested, you need to click the `Resolve conversation` button here as a confirmation.

MilosPaunovic reviewed (changes requested):
Hello @mohammadshahidbeigh,

Before me taking a proper look at the FlowConcurrency.vue component, you need to revert changes from the unrelated file.

Thanks for the efforts so far! 🍀

mohammadshahidbeigh committed:
Reverted monaco-editor.d.ts to original state

MilosPaunovic committed:
Merge branch 'develop' into convert-flow-concurrency-to-ts

Piyush-r-bhaskar reviewed (commented):
Hey @mohammadshahidbeigh 

Thanks for this 🚀

Before I take a look. would you play little bit with flow concurrency if everything works as expected as before ?

Piyush-r-bhaskar commented on ui/src/components/flows/FlowConcurrency.vue:
```diff
@@ -32,51 +32,70 @@
+    }>()
+
+    const flowStore = useFlowStore()
+    type FlowLike = { id?: string; namespace?: string; concurrency?: { limit?: number; behavior?: string } }
```
we try to avoid type. why not interface

mohammadshahidbeigh replied:
✅ Converted type to interface:
Changed type FlowLike to interface FlowLike as requested
Also converted type RunningCountPayload to interface RunningCountPayload for consistency

Piyush-r-bhaskar commented on ui/src/components/flows/FlowConcurrency.vue:
```diff
@@ -32,51 +32,70 @@
+        const raw = Number(runningCount.value)
+        const safe = Number.isFinite(raw) && raw > 0 ? raw : 0
+        const clamped = Math.max(0, Math.min(safe, max))
+        return Math.min(100, (clamped / max) * 100)
```
could you make the code more maintenanble for us ..one step would be to group the `computed` , `functions`

mohammadshahidbeigh replied:
✅ Improved code organization and maintainability:
Added clear section comments to group related code:
// Interfaces - All interface definitions
// Type guards - Type checking functions
// Reactive state - All ref declarations
// Computed properties - All computed values
// Functions - All function definitions

Piyush-r-bhaskar commented on ui/src/components/flows/FlowConcurrency.vue:
```diff
@@ -32,51 +32,70 @@
-        },
-        emits: ["expand-subflow"],
-        data() {
-            return {
-                runningCount: 0,
-                totalCount: 0,
-                runningCountSet: false,
-            }
-        },
```
can we avoid so much variable assignment ?

mohammadshahidbeigh replied:
✅ Reduced variable assignments:
Simplified the setRunningCount function by using chained assignment: runningCount.value = totalCount.value = count instead of separate assignments

mohammadshahidbeigh committed:
refactor(flows): enhance FlowConcurrency.vue with TypeScript interfaces and improved type guards

- Introduced FlowLike and RunningCountPayload interfaces for better type safety.
- Refactored setRunningCount function to streamline handling of running counts.
- Improved code organization with clear sections for interfaces, type guards, and functions.

mohammadshahidbeigh committed:
Merge branch 'convert-flow-concurrency-to-ts' of https://github.com/mohammadshahidbeigh/kestra into convert-flow-concurrency-to-ts

MilosPaunovic commented:
Is this now ready for another review @mohammadshahidbeigh?

MilosPaunovic committed:
Merge branch 'develop' into convert-flow-concurrency-to-ts

MilosPaunovic committed:
Merge branch 'develop' into convert-flow-concurrency-to-ts

mohammadshahidbeigh commented:
Yes!

MilosPaunovic commented:
@Piyush-r-bhaskar, give this a look when you get a chance.

MilosPaunovic committed:
Merge branch 'develop' into convert-flow-concurrency-to-ts

Piyush-r-bhaskar committed:
fix: convert it properly

Piyush-r-bhaskar committed:
Merge branch 'develop' into convert-flow-concurrency-to-ts
